pkg/kubelet/kubelet.go

```diff
@@ -328,6 +328,41 @@ func getRuntimeAndImageServices(remoteRuntimeEndpoint string, remoteImageEndpoin
 	return rs, is, err
 }
 
+func initCertificateManager(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
+	kubeDeps *Dependencies,
+	klet *Kubelet,
+	cloudIPs []net.IP,
+	cloudNames []string,
+	hostnameOverride,
+	certDirectory string) (err error) {
+	var ips []net.IP
+	cfgAddress := net.ParseIP(kubeCfg.Address)
+	if cfgAddress == nil || cfgAddress.IsUnspecified() {
+		localIPs, err := allLocalIPsWithoutLoopback()
+		if err != nil {
+			return err
+		}
+		ips = localIPs
+	} else {
+		ips = []net.IP{cfgAddress}
+	}
+
+	ips = append(ips, cloudIPs...)
```

Comment:
I think a better factoring would be to just move this chunk into a "getKubeletIPs" function.